code/modules/surgery/organs/external/_external_organs.dm

```diff
@@ -250,13 +257,25 @@
 				return
 			var/mob/living/carbon/human/human_owner = ownerlimb.owner
 			draw_color = human_owner.hair_color
-	color = draw_color
+	//Fuck giving the whole thing the same color, instead every layer can have its own
+	if(apply_color_to_layer)
+		layer_to_color_index = list()
+		for(var/draw_layer in all_layers)
+			if(draw_layer & layers)
+				//We make our layer a string because it'll runtime if we index with integers out of sequence
+				layer_to_color_index["[bitflag_to_layer(draw_layer)]"] = get_overlay_color(draw_layer)
+	else
+		color = draw_color
```

Comment:
wait why does layer_to_color_index exist if get_overlay_color exists

Reply:
The idea was to cache them, both so we controle when they change and to help with performance. Although I can just generate them with an overridable proc using just the draw_color and layer